core/src/duration.rs

```diff
@@ -0,0 +1,100 @@
+use num_traits::Zero;
+use ordered_float::OrderedFloat;
+use serde::{Deserialize, Serialize};
+use std::ops::{Add, AddAssign, Sub, SubAssign};
+use std::time::Duration;
+
+/// Duration f64 nanosec
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Default, Serialize, Deserialize)]
+pub struct RuffleDuration(OrderedFloat<f64>);
```

Comment:
For ease of typing and to follow Rust naming conventions, let's name this `Duration` instead of `RuffleDuration`. Rust prefers to avoid stuttering in names (`ruffle_core::RuffleDuration` vs. `ruffle_core::Duration`). We can explicitly specify for methods that want the `std` version (`std::instant::Duration`), or use a renaming import (`use std::instant::Duration as StdDuration`).

Instead of adding `ordered_float` dependency so that we can derive `Ord` and `Eq`, let's instead implement any necessary methods ourselves. For example, `Ord` isn't necessary if we add a direct `min`/`max` method:
```
    pub fn min(self, other: RuffleDuration) -> RuffleDuration {
        if self.0 < other.0 {
            self
        } else {
            other
        }
    }
```
then the duration can become:
`pub struct Duration(f64);`
without the need for `ordered_float`.

I know we already have lots of dependencies, but I like to avoid adding new ones when possible.